clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/spark/datawriter/jobs/DataWriterSparkJob.java

```diff
@@ -55,4 +85,53 @@ protected Dataset<Row> getUserInputDataFrame() {
     }
     return dataFrameReader.load();
   }
+
+  private Dataset<Row> getAvroDataFrame(SparkSession sparkSession, PushJobSetting pushJobSetting) {
+    Dataset<Row> df =
+        sparkSession.read().format("avro").option("pathGlobFilter", GLOB_FILTER_PATTERN).load(pushJobSetting.inputURI);
+
+    // Transforming the input data format
+    df = df.map((MapFunction<Row, Row>) (record) -> {
+      Schema updateSchema = null;
+      if (pushJobSetting.generatePartialUpdateRecordFromInput) {
+        updateSchema = AvroCompatibilityHelper.parse(pushJobSetting.valueSchemaString);
+      }
+
+      GenericRecord rowRecord = RowToAvroConverter.convert(record, pushJobSetting.inputDataSchema);
+      VeniceAvroRecordReader recordReader = new VeniceAvroRecordReader(
+          pushJobSetting.inputDataSchema,
+          pushJobSetting.keyField,
+          pushJobSetting.valueField,
+          pushJobSetting.etlValueSchemaTransformation,
+          updateSchema);
+
+      AvroWrapper<IndexedRecord> recordAvroWrapper = new AvroWrapper<>(rowRecord);
+      final byte[] inputKeyBytes = recordReader.getKeyBytes(recordAvroWrapper, null);
+      final byte[] inputValueBytes = recordReader.getValueBytes(recordAvroWrapper, null);
+
+      return new GenericRowWithSchema(new Object[] { inputKeyBytes, inputValueBytes }, DEFAULT_SCHEMA);
+    }, RowEncoder.apply(DEFAULT_SCHEMA));
+
+    return df;
+  }
+
+  @Deprecated
```

Comment:
I wonder whether we can drop the vson support. We are adding a new method that's marked as deprecated right away

Reply:
There are some stores that still use Vson input